src/OrbitSshQt/include/OrbitSshQt/StateMachineHelper.h

```diff
@@ -193,38 +196,35 @@ class StateMachineHelper : public QObject {
   }
 
   [[nodiscard]] bool IsStarting() const {
-    return state_ >= State::kInitial && state_ < State::kStarted;
+    return state_ >= State::kStarting && state_ < State::kStarted;
   }
 
   [[nodiscard]] bool IsStarted() const {
-    return state_ >= State::kStarted && state_ < State::kShutdown;
+    return state_ >= State::kStarted && state_ < State::kStopping;
   }
 
-  [[nodiscard]] static State PreviousState(State state) {
-    return static_cast<State>(static_cast<std::underlying_type_t<State>>(state) - 1);
+  [[nodiscard]] bool IsStopping() const {
+    return state_ >= State::kStopping && state_ < State::kStopped;
   }
 
-  [[nodiscard]] bool IsStopping() const {
-    return state_ >= State::kShutdown && state_ < PreviousState(State::kError);
+  [[nodiscard]] bool IsStopped() const {
+    return state_ >= State::kStopped && state_ < State::kError;
   }
 
-  [[nodiscard]] bool IsStopped() const { return state_ == PreviousState(State::kError); }
   [[nodiscard]] bool IsInErrorState() const { return state_ == State::kError; }
 
  private:
   friend Derived;
 
-  Derived* self() { return static_cast<Derived*>(this); }
-  const Derived* self() const { return static_cast<const Derived*>(this); }
+  [[nodiscard]] Derived* self() { return static_cast<Derived*>(this); }
+  [[nodiscard]] const Derived* self() const { return static_cast<const Derived*>(this); }
 
-  State state_ = State::kInitial;
+  State state_ = State::kInitialized;
 
-  State CurrentState() const { return state_; }
-  void SetStateHook(State) {}
+  [[nodiscard]] State CurrentState() const { return state_; }
   void SetState(State state) {
     if (state_ != state) {
-      self()->SetStateHook(state);
-      if (state != State::kError && state >= State::kShutdown && state_ < State::kShutdown) {
+      if (state != State::kError && state >= State::kStopping && state_ < State::kError) {
```

Comment:
Note that there has been a real bug here. The last mentioned state should have always been `kError`.